### PR TITLE
apple: Add support for MacBook Air 3,X

### PR DESCRIPTION
**Pull request by ilian**
Add support for MacBook Air 3,X, which only contains a NVIDIA GeForce 320M GPU instead of integrated Intel graphics.
The configuration contains the [black screen fix](https://askubuntu.com/questions/264247/proprietary-nvidia-drivers-with-efi-on-mac-to-prevent-overheating/613573#613573) for this model.

### CODE REVIEW

File: apple/macbook-air/3/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    device = lib.mkDefault "nodev";
+    extraConfig = lib.mkDefault ''
+      insmod setpci;
+      setpci -s "00:17.0" 3e.b=8;
+      setpci -s "02:00.0" 04.b=7;
+    '';
+  };
+
+  # Requires nixpkgs.config.allowUnfree = true;
+  services.xserver.videoDrivers = [ "nvidiaLegacy340" ];
```
[the remark this answers is not in the document]
Comment (Mic92) [reply apple/macbook-air/3/default.nix]: Lately I have seen the performance to be unusable slow on a system with nouveau and graphic card of this generation.

[the remark this answers is not in the document]
Comment (kalbasit) [reply apple/macbook-air/3/default.nix]: Accepting this PR forces users of this laptop to have unfree set to true, not sure if it is the right move here, what do you think @Mic92?

[the remark this answers is not in the document]
Comment (yegortimoshenko) [reply apple/macbook-air/3/default.nix]: Generally would prefer a `mkIf nixpkgs.config.allowUnfree` conditional, but if @Mic92 reports unusable performance, there's not much point anyway.

MacBooks generally require proprietary blobs for complete functionality (say, `facetimehd` is usually required for webcam, and non-free Broadcom driver is required to make Wi-Fi work).

**Comment by nocent** [comment]
Hi, thanks for reviewing!
I've tested the current state of the video driver support of this model:
Starting display-manager from the install ISO results in X Server not finding a screen and thus requires installation from the tty.
After installation, nouveau seems to be working fine out of the box and does not require `setpci` tweaking. KDE desktop effects and video playback at 1080p@60fps of an H.264 sample file are both working without requiring the installation of the proprietary Nvidia driver.
Installing the legacy Nvidia driver causes a kernel panic as soon as the X server starts even though it worked fine about 6 months ago.
```
[  61.431767] BUG: unable to handle kernel NULL pointer dereference at 0000000000000328
[  61.431950] IP: _nv006854rm+0x272/0x3c0 [nvidia]
```
I'm closing this PR because nouveau is working well and seems stable, while the proprietary driver is breaking.

**Comment by Mic92** [comment]
It would probably still make sense to have this module though.
Having it explicit mentioned also it just imports an upper profile
is a good indication for other people that the module was 
tested for this device.

File: apple/macbook-air/3/default.nix [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    ../../.
+    ../../../common/pc/laptop
+    ../../../common/pc/ssd
+  ];
+
+  # Built-in iSight is recognized by the generic uvcvideo kernel module
+  hardware.facetimehd.enable = false;
+
+  hardware.cpu.intel.updateMicrocode =
+    lib.mkDefault config.hardware.enableRedistributableFirmware;
```
Comment (Mic92) [inline apple/macbook-air/3/default.nix]: Why not using `../../../common/cpu/intel` here?

Comment (Mic92) [reply apple/macbook-air/3/default.nix]: Ah, just read about the `NVIDIA GeForce 320M GPU`

**Pull request merged**